clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10319,6 +10319,19 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
   if (Args.hasArg(options::OPT_v))
     CmdArgs.push_back("--wrapper-verbose");
 
+  // Pass the device triple to the linker wrapper tool for SYCL offload.
+  // Only spir64 is currently passed.
+  // TODO(NOM1): Support target triples in a more generic way.
+  // TODO(NOM3): Investigate why passing spir64-unknown-unknown does not work.
+  CmdArgs.push_back("--triple=spir64");
+
```

Comment:
Could you clarify why we need to pass the offload triple? I thought the triple was being retrieved from the object file. Is this a leftover artifact from debugging?

Reply:
Hi @bader 

I was considering the scenario where we have multiple fat objects with multiple target triples. Also, we have device library files that will have their triples. In such a case, we might end up with a large number of triples and might end up compiling for all. Passing in the triple makes sure we are accumulating only those object files that contain that triple and do compilation only for them.
Please let me know if I missed something.

Thanks

Reply:
This sounds like an optimization which is premature for now. If an object file was compiled for multiple offload triples, I think the user expects all of them to be linked. I don't think we should drop everything except `spir64` triple.

Reply:
Hmm. Thinking about this, I think this is more of a limitation of my current implementation. I have added a TODO to improve on this in subsequent PRs. Maybe I can clearly state that we will also consider scenarios where user does not send in the triple.

Thanks

Reply:
The object files may be compiled with many targets but  user may  link with a subset of targets used during obj generation so we should only build an image for which the  user requested.